### ISSUE TITLE
clouddriver HA breaks in 1.20.0 upgrade

### ISSUE THREAD

**Issue by jmymy** [spinnaker/spinnaker#5736]
### Issue Summary:
https://github.com/spinnaker/clouddriver/commit/4831979f725312aaa1c84d268ceb0af46b8e3445
changes the default behavior for the clouddriver-ro redis endpoint. It no longer uses the default redis one and requires an explicit definition.

It appears Halyard is not generating the services.redisRo entry in spinnaker.yml 

```
Caused by: java.net.URISyntaxException: Illegal character in path at index 1: ${services.redisRo.baseUrl}
        at java.base/java.net.URI$Parser.fail(URI.java:2913) ~[na:na]
        at java.base/java.net.URI$Parser.checkChars(URI.java:3084) ~[na:na]
        at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3166) ~[na:na]
        at java.base/java.net.URI$Parser.parse(URI.java:3125) ~[na:na]
        at java.base/java.net.URI.<init>(URI.java:600) ~[na:na]
        at java.base/java.net.URI.create(URI.java:881) ~[na:na]
        ... 150 common frames omitted
```

### Cloud Provider(s):
GCP Kubernetes

### Environment:
Spinnaker deployed into GCP via https://github.com/GoogleCloudPlatform/spinnaker-for-gcp/
Halyard 1.35.3 (gcr.io/spinnaker-marketplace/halyard:1.35.3)

### Feature Area:
Clouddriver

### Description:

- halyard not generating the correct config in spinnaker.yml

- Should have a Breaking change section in the release notes about this behavior?? or add back in the default behavior.

- update documentation https://www.spinnaker.io/reference/halyard/high-availability/ to state that clouddriver-ro no longer uses the default redis.

### Steps to Reproduce:
Setup spinnaker HA by just using `hal config deploy ha clouddriver enable` without passing in the parameters for each redis (--redis-master-endpoint, etc)

### Additional Details:

(edit) after more digging, It actually appears that halyard is not generating the redisRo entry in the spinnaker.yml files

(edit) Confirmed editing the spinnaker.yml secret for clouddriver-ro.yml and adding in a redisRo service, allows clouddriver-ro to startup successfully

---

**Comment by ethanfrogers** [comment]
@ezimanyi was 1.20 supposed to require a newer version of halyard since the changes made [here](https://github.com/spinnaker/clouddriver/commit/4831979f725312aaa1c84d268ceb0af46b8e3445) assumed changed in a generated file?

**Comment by ezimanyi** [comment]
The issue was just a missing default back to the regular redis endpoint; fixed in the above PR and will backport to 1.20.

**Comment by jmymy** [comment]
Thanks for the quick resolution @ezimanyi !
